Include/httpClient/types.h

```diff
@@ -81,3 +81,50 @@
 typedef uint32_t HC_MEMORY_TYPE;
 typedef struct HC_CALL* HC_CALL_HANDLE;
 typedef uint64_t HC_TASK_HANDLE;
+
+typedef enum HC_RESULT
+{
+    HC_OK = 0,
+    HC_E_FAIL = -1,
+    HC_E_POINTER = -2,
+    HC_E_INVALIDARG = -3,
+    HC_E_OUTOFMEMORY = -4,
+    HC_E_BUFFERTOOSMALL = -5,
+    HC_E_NOTINITIALISED = -6,
+    HC_E_FEATURENOTPRESENT = -7,
+} HC_RESULT;
+
+
+#define CONVERT_STD_EXCEPTION(operation) \
```

Comment:
Instead of using these macros around places that we think will throw, I would suggest making sure that all api entry points have a try catch block that translates exceptions at that level (you can look at how I set up Xal for an example).

The advantage is that it's a lot easier to audit the code to be sure that no exceptions leak.
I've even marked all my C boundary functions with noexcept so that the compiler will emit a crash if an exception were to leave for an extra layer of safety (well determinism anyway)

Reply:
I'll look into this in a future PR